### PR TITLE
colorama stubs

### PR DESCRIPTION
**Pull request by hatal175**
As I've noticed you usually welcome stubs for pypi packages, I didn't open an issue and just open a pull request as these are new stubs for the colorama package.
I think they are pretty full (not a big package).

Any input is welcome.

### CODE REVIEW

File: stubs/colorama/colorama/ansitowin32.pyi [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from typing_extensions import Protocol
+
+if sys.platform == "win32":
+    from .winterm import WinTerm
+
+    winterm: WinTerm
+else:
+    winterm: None
+
+class SupportsWrite(Protocol):
```
Comment (Akuli) [inline stubs/colorama/colorama/ansitowin32.pyi:12]: You could use `from _typeshed import SupportsWrite`

Comment (hatal175) [reply stubs/colorama/colorama/ansitowin32.pyi]: done

**Comment by hatal175** [comment]
Um, actually I looked at the code again. There's a class called AnsiCodes. The way it's built, the class instance has its attributes as int but every specific instance has them as string (converted in __init__. How would you suggest handling this?

**Comment by Akuli** [comment]
I would add a comment about it and let them be strings, since most people presumably want to use them as strings. @srittau @hauntsaninja any thoughts?

```
>>> from colorama.ansi import AnsiFore
>>> AnsiFore.CYAN
36
>>> AnsiFore().CYAN
'\x1b[36m'
```

File: stubs/colorama/colorama/initialise.pyi [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+orig_stdout: Optional[TextIO]
+orig_stderr: Optional[TextIO]
+wrapped_stdout: Union[TextIO, StreamWrapper]
+wrapped_stderr: Union[TextIO, StreamWrapper]
+atexit_done: bool
+
+def reset_all() -> None: ...
+def init(autoreset: bool = ..., convert: Optional[bool] = ..., strip: Optional[bool] = ..., wrap: bool = ...) -> None: ...
+def deinit() -> None: ...
+def colorama_text(*args: Any, **kwargs: Any) -> None: ...
```
Comment (JelleZijlstra) [inline stubs/colorama/colorama/initialise.pyi:14]: This has @contextmanager
```suggestion
def colorama_text(*args: Any, **kwargs: Any) -> ContextManager[None]: ...
```

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Thanks! I have a few small comments.

For the int/str problem, it looks like users would normally just access the str values, so it makes sense to put those in the types. If it becomes an issue we could use `Any` instead.

**Comment by ssbarnea** [comment]
Did colorama refused to add types directly to it? I kinda prefer to see type hints directly in the libs but even stubs here are far better than nothing, especially for a popular library like colorama.

**Comment by hatal175** [comment]
Well... that's somewhat of a faux pas on my part. Let me ask. sorry for the
trouble.

**Comment by ssbarnea** [comment]
@haakenlid Ask, the answer is not always positive but it would make it easier to maintain. A major stopper was so far support for old pythons but as soon a library starts to require >= py36, it makes sense to add type directly to the code.

**Comment by hatal175** [comment]
There's this:
tarley/colorama#206
which is not really resolved since early 2019.

I defer to your judgement about this. What should I push for?

**Comment by JelleZijlstra** [comment]
It's fine to just add stubs if the library author is not responsive or not willing to add types. In general we want to make it easier to distribute stubs through typeshed.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**